### PR TITLE
fix(ingestion/mongodb): MongoDB source unable to parse datetimes with years > 9999  

### LINKED ISSUE #10079: MongoDB source unable to parse datetimes with years > 9999
**Describe the bug**

This is a classical python error, if the year on the DateTime of MongoDB is larger than `datetime.MAXYEAR` (9999), the pymongo driver returns an error. This is a super easy fix, and the stack trace hints at the it:

> bson.errors.InvalidBSON: year 275760 is out of range (Consider Using CodecOptions(datetime_conversion=DATETIME_AUTO) or MongoClient(datetime_conversion='DATETIME_AUTO')). See: https://pymongo.readthedocs.io/en/stable/examples/datetimes.html#handling-out-of-range-datetimes

**To Reproduce**
1. Create a mongodb document with a datetime field with year > 9999
2. Ingest data from this MongoDB collection
3. See error

**Expected behavior**
DataHub should be able to parse documents with unreasonably large years.

**Additional context**

I'm on version `0.13.1`.

I think the fix would be changing `datahub/ingestion/source/mongodb.py` line 285, and adding the following:

```python
self.mongo_client = pymongo.MongoClient(self.config.connect_uri,datetime_conversion='DATETIME_AUTO', **options)  # type: ignore
```

Sorry that I don't have time now to submit a PR. If nobody gets to this before, I can try doing it in the weekend. Thank you for DataHub, I really love it

### PR DESCRIPTION
**Pull request by jonasHanhan**
… years > 9999

Closes #10079 

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/mongodb.py [hunk 1]
```diff
@@ -282,7 +282,8 @@ def __init__(self, ctx: PipelineContext, config: MongoDBConfig):
             **self.config.options,
         }
 
-        self.mongo_client = pymongo.MongoClient(self.config.connect_uri, **options)  # type: ignore
+        self.mongo_client = pymongo.MongoClient(self.config.connect_uri, datetime_conversion='DATETIME_AUTO',
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/mongodb.py:285]: let's add a link to https://pymongo.readthedocs.io/en/stable/examples/datetimes.html#handling-out-of-range-datetimes as a comment

**Review by hsheth2 — COMMENTED** [review]
one quick suggestion, and also it seems like the linter isn't happy

looks good other than that

File: metadata-ingestion/src/datahub/ingestion/source/mongodb.py (hunk 1 above)
Comment (jonasHanhan) [reply metadata-ingestion/src/datahub/ingestion/source/mongodb.py:285]: I took a good look at this link, it was indeed my problem, and I'm very sorry that it was my negligence

File: metadata-ingestion/src/datahub/ingestion/source/mongodb.py [hunk 2]
```diff
@@ -281,8 +281,11 @@ def __init__(self, ctx: PipelineContext, config: MongoDBConfig):
             **options,
             **self.config.options,
         }
+        codec_options: CodecOptions = CodecOptions(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/mongodb.py:284]:
```suggestion
        # See https://pymongo.readthedocs.io/en/stable/examples/datetimes.html#handling-out-of-range-datetimes
        codec_options: CodecOptions = CodecOptions(
```

**Review by hsheth2 — APPROVED** [review]

**Comment by hsheth2** [comment]
Looks like CI is failing
```
FAILED tests/integration/mongodb/test_mongodb.py::test_mongodb_ingest - datahub.ingestion.run.pipeline.PipelineInitError: Failed to configure the source (mongodb): Unknown option codec_options
```

**Pull request merged**